plugins-api/src/main/scala/za/co/absa/enceladus/plugins/api/postprocessor/PostProcessor.scala

```diff
@@ -27,9 +27,9 @@ abstract class PostProcessor extends Plugin {
    * This callback function will be invoked when the output data is ready.
    *
    * @param dataFrame A DataFrame containing the output data.
-   * @param params Additional key/value parameters provided by Enceladus.
+   * @param params Additional parameters provided by Enceladus.
    * @return A dataframe with post processing applied
    */
-  def onDataReady(dataFrame: DataFrame, params: Map[String, String]): DataFrame
+  def onDataReady(dataFrame: DataFrame, params: PostProcessorPluginParams): DataFrame
```

Comment:
The idea behind the Map is to be flexible with supporting various post-processing plugins, not only ones that send error columns to message queues.
This change will make the interface more rigid since if/when more plugins are written the need to be coupled with the specific versions of Enceladus. Changes to `PostProcessorPluginParams` will make previously written plugins incompatible until they update Enceladus plugins API dependency.

Reply:
Thanks for explaining, I agree that it is more versatile like that.

To make things bearable (instead of having to hardcode keys everywhere), the `PostProcessorPluginParams` in `plugins-builtin` now has auxiliary methods `fromMap` and `toMap` to easily transform to a flexible Map to a value class and vice-versa.

The API package, thus, remains unpolluted with specific builtin plugins implementation details. 🎉